src/AerosolModel.jl

```diff
@@ -121,7 +119,7 @@ function Mode_κ(
 end
 
 """ number of components in the mode """
-n_components(::Mode_κ{NCOMP}) where {NCOMP} = NCOMP
+n_components(::Mode_κ{T}) where {T} = fieldcount(T)
```

Comment:
This returns 0 when T is any numerical type (and not a tuple). This is why aerosol activation test fails. Because at line 91 of AerosolActivation.jl this function is used and return 0 and thus mean hygroscopicity becomes zero and critical supersaturation becomes inf etc.